### DISCUSSION TITLE
How to change text on bbox visualizer?

### DISCUSSION THREAD

**Discussion by nadyadtm** [facebookresearch/detectron2#3627] [Q&A] [answered]
I want to change the prediction text (ex: class 100% to class) in visualizer bbox. How can I change it in detectron2?

**Comment by sukjunhwang** [comment, marked as answer]
This is probably the function you should edit.

https://github.com/facebookresearch/detectron2/blob/d1e04565d3bec8719335b88be9e9b961bf3ec464/detectron2/utils/visualizer.py#L230

**Reply by nadyadtm** [reply]
Okay, I'm sorry I'm still newbie on Detectron2, so how do I run the code since visualizer is the default function in Detectron2? Do I have to re-install detectron again?

**Reply by sukjunhwang** [reply]
Hi @nadyadtm ,

If you are working on visualizing object detection or instance segmentation predictions on images, this function below is where Visualizer calls _create_text_labels.

https://github.com/facebookresearch/detectron2/blob/d1e04565d3bec8719335b88be9e9b961bf3ec464/detectron2/utils/visualizer.py#L383-L398

_create_text_labels simply outputs texts to be shown on top of bounding boxes.
So, if you want a different format of texts to be written, you can either fix that function directly, or create a class inheriting Visualizer and fix as whatever you want to.

**Reply by nadyadtm** [reply]
Sorry for the late reply

So that means I just call the function as usual after I change the function, right? No need to reinstall detectron, right? How I compile the changed function? I'm still confused about it

**Reply by sukjunhwang** [reply]
setup.py is mostly for environment configurations and installation of necessary libraries, so only changing the function would not need further compilation or reinstallation.

**Reply by nadyadtm** [reply]
Okay I'll try it, thanks for the answer
